### PR TITLE
plan: Fix re-install when sha1 differs

### PR DESCRIPTION
**Pull request by mingwandroid**
Hi @mcg1969, I'm not sure this is the right fix, it could be that find_new_location is meant to be returning something different (possibly not doing the dirname?).

### CODE REVIEW

File: conda/plan.py [hunk 1]
```diff
@@ -290,6 +290,9 @@ def ensure_linked_actions(dists, prefix, index=None, force=False, always_copy=Fa
         if not extracted_in:
             actions[inst.EXTRACT].append(dist)
 
+        # find_new_location returns either a `pkgs` directory or the filename.
+        if not fetched_in.endswith('.tar.bz2'):
+            fetched_in = join(fetched_in, install._dist2filename(dist))
```
Comment (mcg1969) [inline conda/plan.py]: I'm not convinced this is correct. `fetched_in` is supposed to return a `.tar.bz2` filename or None if the file doesn't exist. if it's not, there's a bug somewhere else, and we need to find it.

Comment (mcg1969) [inline conda/plan.py]: Oh wait, I see, `find_new_location` may be broken. Hold on

Comment (mcg1969) [inline conda/plan.py]: Yeah, we should fix `find_new_location`. It should be consistent about what it returns.

Comment (mingwandroid) [inline conda/plan.py]: Yeah, I thought that may be the case, but I didn't want to change something called from so many sites.

**Comment by mingwandroid** [comment]
For your consideration @mcg1969.

**Comment by mcg1969** [comment]
This definitely looks better. If it still works both for normal downloads and SHA1 replacements LGTM. Thanks!

**Comment by mingwandroid** [comment]
Seems to be working fine, yeah. @kalefranz, care to merge this?

**Comment by kalefranz** [comment]
Uggg is it really true our test suite doesn't hit these lines at all?  I'm actually not sure... It might be doing it in a subprocess and coverage isn't finding it.  I'll investigate at some point.  Not going to let this block right now though.

**Comment by mingwandroid** [comment]
Yeah, I think we need to look at our testsuite coverage when we can.

**Pull request merged**